src/Scripting/Core/Hosting/Resolvers/RuntimeMetadataReferenceResolver.cs

```diff
@@ -106,28 +131,90 @@ public override ImmutableArray<PortableExecutableReference> ResolveReference(str
             {
                 if (PathResolver != null)
                 {
-                    var resolvedPath = PathResolver.ResolvePath(reference, baseFilePath);
+                    string resolvedPath = PathResolver.ResolvePath(reference, baseFilePath);
                     if (resolvedPath != null)
                     {
                         return ImmutableArray.Create(_fileReferenceProvider(resolvedPath, properties));
                     }
                 }
             }
-            else if (GacFileResolver != null)
+            else
             {
-                var path = GacFileResolver.Resolve(reference);
-                if (path != null)
+                if (GacFileResolver != null)
+                {
+                    string path = GacFileResolver.Resolve(reference);
+                    if (path != null)
+                    {
+                        return ImmutableArray.Create(_fileReferenceProvider(path, properties));
+                    }
+                }
+
+                if (_useCoreResolver && AssemblyIdentity.TryParseDisplayName(reference, out var identity, out var identityParts))
                 {
-                    return ImmutableArray.Create(_fileReferenceProvider(path, properties));
+                    var result = ResolveTrustedPlatformAssemblyCore(identity.Name, properties);
+                    if (result != null)
+                    {
+                        return ImmutableArray.Create(result);
+                    }
                 }
             }
 
             return ImmutableArray<PortableExecutableReference>.Empty;
         }
 
+        private PortableExecutableReference ResolveTrustedPlatformAssemblyCore(string name, MetadataReferenceProperties properties)
+        {
+            if (_lazyTrustedPlatformAssemblies == null)
+            {
+                _lazyTrustedPlatformAssemblies = GetTrustedPlatformAssemblyMap();
+            }
+
+            if (_lazyTrustedPlatformAssemblies.TryGetValue(name, out string path) && File.Exists(path))
+            {
+                return MetadataReference.CreateFromFile(path, properties);
+            }
+
+            return null;
+        }
+
+        private static ImmutableDictionary<string, string> GetTrustedPlatformAssemblyMap()
+        {
+            var set = ImmutableDictionary.CreateBuilder<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            if (CoreClrShim.AppContext.GetData?.Invoke("TRUSTED_PLATFORM_ASSEMBLIES") is string paths)
```

Comment:
Can you provide some documentation on this value?  I have an idea of what it does but I want to read up more about it.  It's possible this may help me with some code in the test caching framework.